### ISSUE TITLE
How to pass null/None to Procedure?

### ISSUE THREAD

**Issue by legomind** [fsprojects/SQLProvider#158]
```sql
CREATE PROCEDURE dbo.test (
	@OptionalParam INT = NULL
)
AS 

SELECT 1
```

```fsharp
//Snip...
[ for r in db.Procedures.test.Invoke(None).ResultSet
                  -> r.MapTo<TestResult> () ]
//Error: Expression was expected to have type int but here has type Option<int>
```

**Comment by colinbull** [comment]
You can't do this as the F# type system doesn't allow you to represent null as int. In cases like this I guess we could detect if the parameter is nullable (presuming the metadata is available) then the type of the parameter would be `Nullable<int>` which I think would cover your needs.

**Comment by pezipink** [comment]
yes this should be pretty easy to do, we will get it in for the coming V1.  It might be nice to put sprocs under their schemas as well, on an unrelated note. They can clash with each other at the moment.

**Comment by legomind** [comment]
Cool. Looking forward to v1

**Comment by xenocons** [comment]
additionally, it might not be possible but if you set the stored procedure to have Null as a default parameter, and you pass null into the sproc, it will send 'default' to the sql, which uses the default value and all is ok

**Comment by JamesFaix** [comment]
Any updates on this?
